File: app/src/main/java/com/example/android/popularmovies/database/FavouriteMovieDatabase.java
package com.example.android.popularmovies.database;


import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {MovieEntity.class}, version = 5, exportSchema = false)
@TypeConverters({ReviewsConverter.class,  VideoUrlsConverter.class, PosterUriConverter.class})
public abstract class FavouriteMovieDatabase extends RoomDatabase {

    private static final String DATABASE_NAME = "FAVOURITE_MOVIES";
    private static final Object LOCK = new Object();
    private static FavouriteMovieDatabase instance;

    public static FavouriteMovieDatabase getInstance(Context context){
        if(instance == null){
            synchronized (LOCK){
                instance = Room.databaseBuilder(context.getApplicationContext(),
                        FavouriteMovieDatabase.class, FavouriteMovieDatabase.DATABASE_NAME)
                        .fallbackToDestructiveMigration()
                        .build();
            }
        }
        return instance;
    }

    public abstract MovieDao movieDao();

}

File: app/src/main/java/com/example/android/popularmovies/database/FavouriteMovieDatabase.kt
package com.example.android.popularmovies.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.android.popularmovies.models.Movie
import com.example.android.popularmovies.models.Review
import com.example.android.popularmovies.models.Video


@Database(entities = [Movie::class, Review::class, Video::class], version = 11, exportSchema = false)
abstract class FavouriteMovieDatabase : RoomDatabase() {
    abstract fun movieDao(): MovieDao

    companion object {
        private const val DATABASE_NAME = "FAVOURITE_MOVIES"
        private val LOCK = Any()

        @Volatile
        private var instance: FavouriteMovieDatabase? = null

        fun getInstance(context: Context): FavouriteMovieDatabase? {
            if (instance == null) {
                synchronized(LOCK) {
                    instance = Room.databaseBuilder(context.applicationContext,
                            FavouriteMovieDatabase::class.java, DATABASE_NAME)
                            .fallbackToDestructiveMigration()
                            .build()
                }
            }
            return instance
        }
    }
}